### PR TITLE
Sync Enhancement. Starts-With and Filter Query

### PR DESCRIPTION
**Pull request by markus-gx**
## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## How to test this PR

```bash
# Sync only stories that starts with `myStartsWithString` from `00001` space to `00002` space
$ storyblok sync --type stories --source 00001 --target 00002 --starts-with myStartsWithString

# Sync only stories with a category field like `reference` from `00001` space to `00002` space
$ storyblok sync --type stories --source 00001 --target 00002 --filter --keys category --operations like --values reference

# Sync only stories with a category field like `reference` and a name field not like `demo` from `00001` space to `00002` space
$ storyblok sync --type stories --source 00001 --target 00002 --filter --keys "category name" --operations "like not_like" --values "reference demo"
```

## What is the new behavior?

Added starts_with parameter and a filter query. Description:

* `starts-with`: sync only stories that starts with the given string
* `filter`: sync stories based on the given filter. Required Options: Required options: `--keys`, `--operations`, `--values`
* `keys`: Multiple keys should be inside quotes and separated by whitespace. Example: `--keys "key1 key2"`, `--keys key1`
* `operations`: Operations to be used for filtering. Can be: `is`, `in`, `not_in`, `like`, `not_like`, `any_in_array`, `all_in_array`, `gt_date`, `lt_date`, `gt_int`, `lt_int`, `gt_float`, `lt_float`. Multiple operations should be inside quotes and separated by whitespace.

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -282,13 +292,36 @@ program
...
+          throw new Error('The number of keys, operations and values must be the same')
+        }
+        const invalidOperators = _operations.filter((o) => !operators.includes(o))
+        if (invalidOperators.length) {
+          throw new Error('Invalid operator(s) applied for filter: ' + invalidOperators.join(' '))
+        }
+        filterQuery = {}
+        _keys.forEach((key, index) => {
+          filterQuery[key] = { [_operations[index]]: _values[index] }
+        })
```
Comment (ademarCardoso) [inline src/cli.js]: I think it's nice to move this code to a util function inside `src/utils` folder, and you could import and use here

Comment (markus-gx) [reply src/cli.js]: Absolutely right! Thanks ;)

**Comment by ademarCardoso** [comment]
@markus-gx I tested and it's woking nice, but please fix the conflict in the `index.js`.
I will pass to the QA of my squad to test another things and when he send a green light we will merge this 😄

**Comment by markus-gx** [comment]
Hey @ademarCardoso ; Are there any updates on this? :)

**Comment by ademarCardoso** [comment]
It is awaiting review of our QA, I believe that by next week it will be reviewed and merged.
cc: @markus-gx

**Comment by carlosericciardi** [comment]
Hello @markus-gx , I performed some tests on the branch, and everything is working correctly. However, I suggest changing the parameter delimiter from whitespace to comma, as some string data may contain whitespaces.

**Comment by markus-gx** [comment]
> Hello @markus-gx , I performed some tests on the branch, and everything is working correctly. However, I suggest changing the parameter delimiter from whitespace to comma, as some string data may contain whitespaces.

Nice to hear ;) 
Which delimiter do you mean? For the filter query? Absolutely right!.

**Comment by ademarCardoso** [comment]
@markus-gx The QA has approved the changes and i will deploy this on Monday 👍🏼   Thanks for the contribution 😄

**Review by ademarCardoso — APPROVED** [review]

**Pull request merged**